### PR TITLE
Quickstart for EKS.

### PR DESCRIPTION
**Pull request by falfaro**

### CODE REVIEW

File: docs/quickstart-eks.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+* [Amazon CLI](https://aws.amazon.com/cli/)
+* [`eksctl`](https://aws.amazon.com/blogs/opensource/eksctl-eks-cluster-one-command/)
+* [Kubernetes CLI](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+* [BKPR installer](install.md)
+* [kubecfg](https://github.com/ksonnet/kubecfg/releases)
+* [jq](https://stedolan.github.io/jq/)
+
+### DNS requirements
+
+In addition to the requirements listed above, a domain name is also required for setting up Ingress endpoints to services running in the cluster. The specified domain name can be a top-level domain (TLD) or a subdomain. In either case you have to manually [set up the NS records](#step-3-configure-domain-registration-records) for the specified TLD or subdomain so as to delegate DNS resolution queries to an Amazon Route 53 hosted zone created and managed by BKPR.  This is required in order to generate valid TLS certificates.
```
Comment (arapulido) [inline docs/quickstart-eks.md]: s/(#step-3-configure-domain-registration-records)/(#step-4-configure-domain-registration-records)/

File: docs/quickstart-eks.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  export BKPR_DNS_ZONE=my-domain.com
+  export AWS_EKS_USER=my-user@example.com
+  export AWS_EKS_CLUSTER=my-eks-cluster
+  export AWS_EKS_K8S_VERSION=1.11
+  ```
+
+  - `BKPR_DNS_ZONE` specifies the DNS suffix for the externally-visible websites and services deployed in the cluster.
+  - `AWS_EKS_USER` specifies the email address used in requests to Let's Encrypt.
+  - `AWS_EKS_CLUSTER` specifies the name of the AKS cluster.
+  - `AWS_EKS_K8S_VERSION` specifies the version of Kubernetes to use for creating the cluster. The [BKPR Kubernetes version support matrix](../README.md#kubernetes-version-support-matrix-for-bkpr-10) lists the base Kubernetes versions supported by BKPR.
```
Comment (arapulido) [inline docs/quickstart-eks.md]: s/(../README.md#kubernetes-version-support-matrix-for-bkpr-10)/(../README.md#kubernetes-version-support-matrix-for-bkpr)

File: docs/quickstart-eks.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+Else, If you are new to using BKPR on EKS, or if you want to create a new User Pool in AWS Cognito, please follow these steps:
+
+1. Browse to the [Cognito](https://console.aws.amazon.com/cognito/) module in the AWS Console
+2. Click on the `Manage User Pools` button
+3. Click on the `Create a user pool` button on the top-right corner
+4. Enter a valid `Pool name` like `eks-test`
+5. Click on the `Step through settings` button
+6. In the `Attributes` section leave all the checkboxes unticked as shown below:
+
+<p align="center"><img src="eks/1-Attributes.png" width=840/></p>
```
Comment (arapulido) [inline docs/quickstart-eks.md]: Please, use markdown for the images as well (```Alt Text```)

File: docs/quickstart-eks.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+  ```bash
+  kubectl get nodes
+  ```
+
+### Step 2: Configure a User Pool in AWS Cognito
+
+In order to authenticate users for applications running atop EKS, BKPR requires a User Pool in AWS Cognito to be configured.
+
+If you already have a working User Pool in AWS Cognito that you would like to use for authenticating users, you will need to retrieve its ID in the form `${AWS_REGION}-${USER_POOL_ID}`, and write it down. Using this `Pool Id`, skip at the end of this section.
```
Comment (arapulido) [inline docs/quickstart-eks.md]: Shall we request at this point to export it as AWS_COGNITO_USER_POOL_ID (as it says below). Also, what is the end of this section? move to the next one? something else?

File: docs/quickstart-eks.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+-(import "https://releases.kubeprod.io/files/v1.2.0/manifests/platforms/eks.jsonnet") {
++(import "https://releases.kubeprod.io/files/v1.3.0/manifests/platforms/eks.jsonnet") {
+  config:: import "kubeprod-autogen.json",
+  // Place your overrides here
+ }
+```
+
+### Step 3: Perform the upgrade
+
+Re-run the `kubeprod install` command, from the [Deploy BKPR](#step-2-deploy-bkpr) step, in the directory containing the existing `kubeprod-autogen.json` and updated `kubeprod-manifest.jsonnet` files.
```
Comment (arapulido) [inline docs/quickstart-eks.md]: s/#step-2-deploy-bkpr/#step-3-deploy-bkpr

File: docs/quickstart-eks.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+### DNS requirements
+
+In addition to the requirements listed above, a domain name is also required for setting up Ingress endpoints to services running in the cluster. The specified domain name can be a top-level domain (TLD) or a subdomain. In either case you have to manually [set up the NS records](#step-3-configure-domain-registration-records) for the specified TLD or subdomain so as to delegate DNS resolution queries to an Amazon Route 53 hosted zone created and managed by BKPR.  This is required in order to generate valid TLS certificates.
+
+## Installation and setup
+
+### Step 1: Set up the cluster
+
+In this section, you will deploy an Amazon Elastic Container Service for Kubernetes (Amazon EKS) cluster using the `eksctl` tool`.
```
Comment (sameersbn) [inline docs/quickstart-eks.md]:
```suggestion
In this section, you will deploy an Amazon Elastic Container Service for Kubernetes (Amazon EKS) cluster using the `eksctl` tool.
```

File: docs/quickstart-eks.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+
+* Make sure you have a working Amazon CLI environment. Please refer to [Configuring the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) if in doubt.
```
Comment (sameersbn) [inline docs/quickstart-eks.md]:
```suggestion
* Make sure you have a working Amazon CLI environment. Refer to [Configuring the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for more information.
```

File: docs/quickstart-eks.md [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+### Step 2: Configure a User Pool in AWS Cognito
+
+In order to authenticate users for applications running atop EKS, BKPR requires a User Pool in AWS Cognito to be configured.
+
+If you already have a working User Pool in AWS Cognito that you would like to use for authenticating users, you will need to retrieve its ID in the form `${AWS_REGION}-${USER_POOL_ID}`, and write it down. Using this `Pool Id`, skip at the end of this section.
+
+Else, If you are new to using BKPR on EKS, or if you want to create a new User Pool in AWS Cognito, please follow these steps:
+
+1. Browse to the [Cognito](https://console.aws.amazon.com/cognito/) module in the AWS Console
+2. Click on the `Manage User Pools` button
```
Comment (sameersbn) [inline docs/quickstart-eks.md]: I suggest using **bold** text when referring to UI elements. **Manager User Pools** in this case. IMO code formatting should generally be used when referring to a `command`, `VARIABLE`, etc.

Side-note: You don't explicitly need to increment numbered lists in markdown. You can write them plainly as

```
1. Step 1
1. Next Step
1. Another Step
```

which will be rendered correctly as shown below

1. Step 1
1. Next Step
1. Another Step

File: docs/quickstart-eks.md [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+3. Click on the `Create a user pool` button on the top-right corner
+4. Enter a valid `Pool name` like `eks-test`
+5. Click on the `Step through settings` button
+6. In the `Attributes` section leave all the checkboxes unticked as shown below:
+
+<p align="center"><img src="eks/1-Attributes.png" width=840/></p>
+
+7. Click on the `Next step` button
```
Comment (sameersbn) [inline docs/quickstart-eks.md]: I think this step should be merged with the previous step as in we can write the previous step as follows:

In the **Attributes** section leave all the check-boxes unticked and go to the **Next step**:

The same suggestion goes for other instances as well

File: docs/quickstart-eks.md [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+  export AWS_COGNITO_USER_POOL_ID=eu-central-1_sHSdWT6VL
+  ```
+
+In order to access protected resources which require authentication like Prometheus, Kibana or Grafana, you will need to configure users in the newly created user pool. The next steps highlight how to create a test user which can be used to access these protected resources:
+
+1. Browse to the [Cognito](https://console.aws.amazon.com/cognito/) module in the AWS Console
+2. Click on the `Manage User Pools` button
+3. Click on the button for the user pool you just created
+4. Click on the `Users and groups` option from the menu on the left
+5. Click on the `Create user` button
```
Comment (sameersbn) [inline docs/quickstart-eks.md]: I would suggest changing this entire sequence to:

2. Navigate to **Manage User Pools > _YOUR_USER_POOL_ > Users and Groups > Create user**

File: docs/quickstart-eks.md [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+In order to authenticate users for applications running atop EKS, BKPR requires a User Pool in AWS Cognito to be configured.
+
+If you already have a working User Pool in AWS Cognito that you would like to use for authenticating users, you will need to retrieve its ID in the form `${AWS_REGION}-${USER_POOL_ID}`, and write it down. Using this `Pool Id`, skip at the end of this section.
+
+Else, If you are new to using BKPR on EKS, or if you want to create a new User Pool in AWS Cognito, please follow these steps:
+
+1. Browse to the [Cognito](https://console.aws.amazon.com/cognito/) module in the AWS Console
+2. Click on the `Manage User Pools` button
+3. Click on the `Create a user pool` button on the top-right corner
+4. Enter a valid `Pool name` like `eks-test`
```
Comment (sameersbn) [inline docs/quickstart-eks.md]: Consider merging these steps into one: f.e.

Navigate to **Manage User Pools > Create a user pool** and specify a **Pool name**, fe. _eks-test_

File: docs/quickstart-eks.md [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+14. Click on the `Next step` button
+15. In the `Devices` section do not change anything and click on the `Next step` button
+16. In the `App clients` section do not change anything and click on the `Next step` button
+17. In the `Triggers` section do not change anything and click on the `Next step` button
+18. In the `Review` section you will see a quick summary of the user pool that will be created, like the onw shown below:
+
+<p align="center"><img src="eks/4-Review.png" width=840/></p>
+
+19. Click on the `Create pool` button to create the user pool
+20. The user pool will be created and the `General settings` page will be displayed, where you can find the User Pool ID and ARN:
```
Comment (sameersbn) [inline docs/quickstart-eks.md]: I suggest merging a lot of these steps

File: docs/quickstart-eks.md (hunk 8 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: I can't use the 1. trick because a paragraph break resets the numbering.

File: docs/quickstart-eks.md (hunk 7 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done.

File: docs/quickstart-eks.md (hunk 6 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done.

File: docs/quickstart-eks.md (hunk 12 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done.

File: docs/quickstart-eks.md (hunk 5 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done.

File: docs/quickstart-eks.md (hunk 3 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: The images look too big when using markdown, hence the reason for pure HTML tags (controlling size).

File: docs/quickstart-eks.md (hunk 2 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done.

File: docs/quickstart-eks.md (hunk 1 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done

File: docs/quickstart-eks.md (hunk 4 above)
Comment (falfaro) [reply docs/quickstart-eks.md]: Done

**Comment by falfaro** [comment]
PTAL

**Comment by falfaro** [comment]
Thx!

**Comment by falfaro** [comment]
bors try

**Comment by sameersbn** [comment]
bors retry

**Pull request merged**